### PR TITLE
WIP - Add messages endpoint

### PR DESCRIPTION
**Pull request by nramadas**
:eyeglasses: @schwers

### CODE REVIEW

File: apis/MessagesEndpoint.es6.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import Comment from '../models2/CommentModel';
+import Post from '../models2/PostModel';
+import Message from '../models2/MessageModel';
+import { runQuery, save } from '../apiBase/APIRequestUtils';
+
+const TYPE_MAP = {
+  't1': Comment,
```
Comment (schwers) [inline apis/MessagesEndpoint.es6.js]: 🔕  I wonder more and more if we should use the constants versions of these. Outside the scope of this though.

File: apis/MessagesEndpoint.es6.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  const { body } = res;
+  body.data.children.forEach(c => apiResponse.addResult(TYPE_MAP[c.kind].fromJSON(c.data)));
+};
+
+const parsePostBody = (res, apiResponse, apiRequest) => {
+  const { body } = res;
+  console.log(body);
+};
+
+const path = (method, query={}) => {
```
Comment (schwers) [inline apis/MessagesEndpoint.es6.js]: Could you change this to `getPath` or `path` for now and only handle the `GET` case? I've found that separating out the method paths has kept things a little cleaner.

File: apis/MessagesEndpoint.es6.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  const { subreddit, type, thingId } = query;
+  if (method === 'get') {
+    const sub = subreddit ? `r/${ subreddit }/` : '';
+    return `${ sub }message/${ type }.json`;
+  }
+  if (!thingId) { return 'api/compose'; }
+  return 'api/comment';
```
Comment (schwers) [inline apis/MessagesEndpoint.es6.js]: Hitting the comment endpoint seems a little ... off here. Will this case just be actual `Message` models or is this intended for `Comments` / `Posts` that are in your inbox?

Comment (nramadas) [inline apis/MessagesEndpoint.es6.js]: honestly, this was a holdover from snoode. i'm not sure when this applies specifically, but I was scared to change it and break something.

File: models2/CommentModel.es6.js [hunk 4]
```diff
@@ -58,14 +58,14 @@ export default class CommentModel extends RedditModel {
     body_html: 'bodyHTML',
     created_utc: 'createdUTC',
     link_id: 'linkId',
-    link_title: 'linkTitle',
```
Comment (schwers) [inline models2/CommentModel.es6.js]: This looks like it was moved accidently

File: models2/MessageModel.es6.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    createdUTC: T.number,
+    subreddit: T.string,
+    parentId: T.string,
+    replies: T.arrayOf(T.string),
+    distinguished: T.string,
+    subject: T.string,
+  };
+
+  static API_ALIASES = {
+    was_comment: 'isComment',
```
Comment (schwers) [inline models2/MessageModel.es6.js]: 🔕 Nice rename, `was_comment` is confusing

**Comment by schwers** [comment]
Just some minor questions, 💅

File: apis/MessagesEndpoint.es6.js (hunk 3 above)
Comment (nramadas) [inline apis/MessagesEndpoint.es6.js]: ah that's good to know. i'll put that comment back in

File: apis/MessagesEndpoint.es6.js (hunk 2 above)
Comment (nramadas) [inline apis/MessagesEndpoint.es6.js]: sure thing

File: models2/CommentModel.es6.js (hunk 4 above)
Comment (nramadas) [inline models2/CommentModel.es6.js]: whoops, good catch

**Comment by nramadas** [comment]
:haircut:

**Comment by schwers** [comment]
:fish:

**Pull request merged**